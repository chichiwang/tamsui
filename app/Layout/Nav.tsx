import React from 'react';
import classNames from 'classnames';

import Logo from 'app/components/Logo';
import ExternalLink from 'app/components/ExternalLink';
import InternalLink from 'app/components/InternalLink';

import styles from './styles.module.scss';

function Nav() {
  return (
    <nav className={styles.navbar}>
      <div className={styles.contents}>
        <div className={classNames(styles.segment, styles.home)}>
          <InternalLink className={styles.menuLink} aria-labelledby="Site-Name" to="/">
            <Logo className={styles.logo} height="30px" />
            <span id="Site-Name">Tamsui</span>
          </InternalLink>
        </div>
        <div className={classNames(styles.segment, styles.pages)}>
          <InternalLink className={styles.menuLink} to="/documentation">
            Documentation
          </InternalLink>
        </div>
        <div className={classNames(styles.segment, styles.social)}>
          <ExternalLink href="https://github.com/chichiwang/tamsui">
            <img alt="Github logo" className={styles.logo} src="/static/images/github-mark.svg" />
          </ExternalLink>
        </div>
      </div>
      <div className={styles.accentBarContainer}>
        <div className={classNames(styles.accentBar, styles.slowSlide)} />
      </div>
    </nav>
  );
}

export default Nav;
