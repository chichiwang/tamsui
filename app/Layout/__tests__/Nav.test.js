import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Nav from '../Nav';

jest.mock('app/components/Logo', function mockedLogo() {
  return function mockLogo() {
    return (
      <div data-testid="mock-logo" />
    );
  };
});

function RouterWrappedNav() {
  return (
    <MemoryRouter>
      <Nav />
    </MemoryRouter>
  );
}

describe('Layout/Nav Component', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<RouterWrappedNav />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('Logo and link home', () => {
    test('renders the site logo', () => {
      render(<RouterWrappedNav />);

      expect(screen.getByTestId('mock-logo')).toBeInTheDocument();
    });

    test('contains the project name', () => {
      render(<RouterWrappedNav />);

      expect(screen.getByText('Tamsui')).toBeInTheDocument();
    });

    test('contains link to root route /', () => {
      render(<RouterWrappedNav />);

      expect(screen.getByRole('link', { name: 'Tamsui' })).toHaveAttribute('href', '/');
    });
  });

  describe('Page Links', () => {
    test('contains a link to the Documentation page', () => {
      render(<RouterWrappedNav />);

      expect(screen.getByText('Documentation')).toBeInTheDocument();
      expect(screen.getByRole('link', { name: 'Documentation' })).toHaveAttribute('href', '/documentation');
    });
  });

  describe('Social Links', () => {
    test('contains a Github logo', () => {
      render(<RouterWrappedNav />);

      expect(screen.getByRole('img', { name: 'Github logo' })).toBeInTheDocument();
    });

    test('contains link to the Github project', () => {
      render(<RouterWrappedNav />);

      expect(screen.getByRole('link', { name: 'Github logo' })).toHaveAttribute('href', 'https://github.com/chichiwang/tamsui');
    });
  });
});
