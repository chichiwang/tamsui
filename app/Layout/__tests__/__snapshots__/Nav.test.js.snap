// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/Nav Component matches snapshot 1`] = `
<nav
  className="navbar"
>
  <div
    className="contents"
  >
    <div
      className="segment home"
    >
      <a
        aria-labelledby="Site-Name"
        className="menuLink"
        href="/"
        onClick={[Function]}
      >
        <div
          data-testid="mock-logo"
        />
        <span
          id="Site-Name"
        >
          Tamsui
        </span>
      </a>
    </div>
    <div
      className="segment pages"
    >
      <a
        className="menuLink"
        href="/documentation"
        onClick={[Function]}
      >
        Documentation
      </a>
    </div>
    <div
      className="segment social"
    >
      <a
        href="https://github.com/chichiwang/tamsui"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Github logo"
          className="logo"
          src="/static/images/github-mark.svg"
        />
      </a>
    </div>
  </div>
  <div
    className="accentBarContainer"
  >
    <div
      className="accentBar slowSlide"
    />
  </div>
</nav>
`;
